### ISSUE TITLE
[Bug]: Updating Angular should be a breaking change

### ISSUE THREAD

**Issue by lorenzodallavecchia** [damienbod/angular-auth-oidc-client#1777]
### Version

14.1.5

### Please provide the exception or error you saw

```true
Upgrading `angular-auth-oidc-client` from 14.0.2. to 14.1.5 I have the following compilation error:

  This application depends upon a library published using Angular version 14.1.0, which requires Angular version 14.0.0 or newer to work correctly.
  Consider upgrading your application to use a more recent version of Angular.
```

### A clear and concise description of what you expected to happen.

```true
My application is still using Angular 13, so the error is correct. However, I was not expecting the error to happen with a minor upgrade of `angular-auth-oidc-client`.
```

### Additional context

The problem happens because Angular is sometimes updated in the source workspace without waiting for a major release.

* Angular 14 in 5ec347912d50afd970ee045c7c98979a68559666 (version 14.1.0)
* Angular 15 in e4640280a0488374eb6f1c73be18e6af1fb7d776 (version 15.0.1)
* Angular 16 in cf8dda00052e73b40d3c0d48de06c5c0d1f343ff (not released - hope we are waiting for 16.0.0 😀)

Unfortunately, Angular Ivy is not forward-compatible, as [explained here](https://github.com/angular/angular/issues/46748#issuecomment-1178817658).

I think it would be better to stick with the Angular version declared as a lower bound in `peerDependencies`. Currently, it is still `12.0.0`. That should not be a big deal because Angular is backward-compatible.

**Comment by lorenzodallavecchia** [comment]
> We are on 16 already. Can you elaborate?

Sorry, I realize now that I referred to the version numbers unclearly.

What I mean is that you have not yet released any `angular-auth-oidc-client` version compiled with Angular 16. The latest released version, 15.0.5, was still compiled with Angular 15.
Since changing the base version of Angular is breaking, it should be released as part of the next _major_ release of `angular-auth-oidc-client`, which would be 16.0.0. Also, the released `package.json` should now require Angular with `>=16.0.0`.

**Comment by FabianGosebrink** [comment]
> What I mean is that you have not yet released any angular-auth-oidc-client version compiled with Angular 16. The latest released version, 15.0.5, was still compiled with Angular 15.

Where do you see this?

**Comment by lorenzodallavecchia** [comment]
I checked commit cf8dda00052e73b40d3c0d48de06c5c0d1f343ff, and it is not yet included in any tagged release.

Another way to see that is by installing `angular-auth-oidc-client@latest` (now `15.0.5`) and looking for Angular AOT metadata inside compiled files.
For example, in `fesm2020/angular-auth-oidc-client.mjs`, towards the end.

```javascript
DefaultLocalStorageService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.0", ngImport: i0, type: DefaultLocalStorageService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
```

The `version: "15.0.0"` refers to the Angular version used for compiling.

By the way, the `minVersion` puzzles me. Previously I took it as an indication that I could use even a version that low (`12.0.0` in this case), but apparently, that's not the case, as I get the compilation error "Consider upgrading your application".

[This discussion](https://github.com/angular/angular/issues/46398) seems to clarify that Angular 14.0.0 broke forward compatibility, and the Angular folks say that it was never intended. It also appears that the `minVersion` field is hard-coded and useless.

> Indeed, the output for libraries contains the exact version used during compilation, as well as a minimum target version. This was baked in from the start of v12 (the initial availability of Ivy's library output) to allow detecting libraries that require a later version at application build time.
>
> In v14, the library output has changed [...] this is a forwards compatibility break, [...]
>
>>    So I wonder if there is any way to instruct the compiler that my library may be used with Angular 12?
>>    Maybe some kind of a metadata that may be given during a build of a library or an app to a compiler/builder?
>
> This is not possible and won't be supported, as there's loads of potential forwards compatibility hazards as Alex mentioned; you'll have to build with the version that you want to support as a minimum.

**Comment by FabianGosebrink** [comment]
That is correct. We will release a version with A16 in the next couple of days.

**Comment by tushar9919** [comment]
Do we have Angular 16 version?

**Comment by timdeschryver** [comment]
You can use [v16.0.0](https://github.com/damienbod/angular-auth-oidc-client/releases/tag/release_16_0_0) @tushar9919
